### PR TITLE
Add AUTOTEST option to CMakeLists.txt

### PR DESCRIPTION
**Pull request by chron0**
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1223)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -371,81 +371,84 @@ unit_test(toxcore util)
...
-  # library into every binary explicitly.
-  #
-  # The alternative is to #include the library in every main file, but I
-  # (@iphydf) felt that this solution was cleaner.
-  add_definitions(-DANDROID_CPU_FEATURES="${ANDROID_CPU_FEATURES}")
-  set(CPUFEATURES other/cpufeatures.c)
-endif()
+option(AUTOTEST "Enable Automated regression tests" OFF)
+if(AUTOTEST)
+  add_library(misc_tools
```
Comment (iphydf) [inline CMakeLists.txt]: misc_tools is needed for other things than auto tests. Better add the condition to the existing `if` inside `function(auto_test target)`.

**Review by iphydf — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -379,73 +379,76 @@ target_link_modules(misc_tools toxcore)
...
 
 if(ANDROID_CPU_FEATURES)
-  # We need to compile cpufeatures.c as many times as there are executables,
-  # because libvpx doesn't include it although it depends on it. We can't get
-  # the link ordering right in cmake, so we need to compile the cpufeatures
-  # library into every binary explicitly.
-  #
-  # The alternative is to #include the library in every main file, but I
-  # (@iphydf) felt that this solution was cleaner.
+    # We need to compile cpufeatures.c as many times as there are executables,
```
Comment (iphydf) [inline CMakeLists.txt]: Undo indentation.

File: CMakeLists.txt [hunk 3]
```diff
@@ -379,73 +379,76 @@ target_link_modules(misc_tools toxcore)
...
-  if(NOT (MSVC AND ARGV1 STREQUAL "MSVC_DONT_BUILD"))
-    add_executable(auto_${target}_test ${CPUFEATURES}
-      auto_tests/${target}_test.c)
-    target_link_modules(auto_${target}_test toxcore misc_tools)
-    if(NOT ARGV1 STREQUAL "DONT_RUN")
-      add_test(NAME ${target} COMMAND ${CROSSCOMPILING_EMULATOR} auto_${target}_test)
-      set_tests_properties(${target} PROPERTIES TIMEOUT "${TEST_TIMEOUT_SECONDS}")
-      set_property(TEST ${target} PROPERTY ENVIRONMENT "LLVM_PROFILE_FILE=${target}.profraw")
+option(AUTOTEST "Enable Automated regression tests" OFF)
+if(AUTOTEST)
```
Comment (iphydf) [inline CMakeLists.txt]: Why not add the condition inside the existing `if`?

**Review by iphydf — CHANGES_REQUESTED** [review]

**Comment by iphydf** [comment]
Can you update README.md with the new option?

**Review by iphydf — APPROVED** [review]

File: CMakeLists.txt [hunk 4]
```diff
@@ -390,8 +390,10 @@ if(ANDROID_CPU_FEATURES)
   set(CPUFEATURES other/cpufeatures.c)
 endif()
 
+option(AUTOTEST "Enable Testing" OFF)
```
Comment (nurupo) [inline CMakeLists.txt]: Would be nice if the description matched the one from INSTALL.md

**Review by nurupo — CHANGES_REQUESTED** [review]
Also enable autotests in https://github.com/TokTok/c-toxcore/blob/master/.travis/cmake-freebsd-stage2#L53

**Review by nurupo — APPROVED** [review]

**Pull request merged**